data/doctests.js

```diff
@@ -22,15 +33,27 @@ var docTests = {
   "emptyElements": {
     name: "empty_elements",
     desc: "empty_elements_desc",
-    check: function checkEmptyElements(content) {
-      var matches = content.match(/<[^\/][^>]*?>([\s\r\n]|&nbsp;|<br\/?>)*<\/.*?>/gi) || [];
-      for (var i = matches.length - 1; i >= 0; i--) {
-        if (matches[i].match(/^<(?:link|track|param|area|command|col|base|meta|hr|source|img|keygen|br|wbr|input)/)) {
-          matches.splice(i, 1);
-        }
+    check: function checkEmptyElements(rootElement) {
+      var treeWalker = document.createTreeWalker(
+          rootElement,
+          NodeFilter.SHOW_ELEMENT,
+          {
+            acceptNode: (node) => {
+              //alert(node.localName + ", " + node.textContent + ", " + node.localName.match(/^(?:link|track|param|area|command|col|base|meta|hr|source|img|keygen|br|wbr|input)$/i));
```

Comment:
I think removing the `alert` and explaining that this excludes self-closing elements is nicer.